### PR TITLE
DOC: added examples to DataFrame.var #44162

### LINKED ISSUE #44162: DOC: var and std missing examples
### 

- [X] I have checked that the issue still exists on the latest versions of the docs on `master` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.var.html
https://pandas.pydata.org/docs/dev/reference/api/pandas.DataFrame.std.html

Screenshot (1178)

### Documentation problem

In all other APIs in DataFrame at least one example is given for better understanding.
I believe, function description is also available in notebooks but not any example.
It's good to have one example for mentioned APIs.

### Suggested fix for documentation

Fix: Use of atleast one example for better undertanding of function.
It's better because function description along with example creates clarity of its use and implimentation.

### PR DESCRIPTION
**Pull request by ruankie**
- [x] closes #44162 by adding two examples to docstring of DataFrame.var
- [x] tests added / passed
- [x] All linting tests passed (pre-commit)

Screenshot of locally built documentation:

var

### CODE REVIEW

**Review by MarcoGorelli — APPROVED** [review]
Nice, thanks @ruankie 

Looks like there's some old commits here, but they'll get squashed together when merging anyway so it's fine

**Comment by ruankie** [comment]
Thanks @MarcoGorelli for the motivation via cat images.
Apologies for the old commits. Next time I'll create a new branch from scratch.

**Comment by ruankie** [comment]
It seems like there is a check that keeps failing `CI > Checks (pull_request) > Typing validation`. I do not understand the error report so I'm not sure what is going wrong. Please see the details of the error report below and advise:

```console
Run ci/code_checks.sh typing
mypy --version
mypy 0.910
Performing static analysis using mypy
pandas/core/frame.py:997: error: Incompatible return value type (got "Optional[str]", expected "str")  [return-value]
Found 1 error in 1 file (checked 1318 source files)
Performing static analysis using mypy DONE
Performing static analysis using pyright
No configuration file found.
pyproject.toml file found at /home/runner/work/pandas/pandas.
Loading pyproject.toml file at /home/runner/work/pandas/pandas/pyproject.toml
Assuming Python platform Linux
Searching for source files
Found 311 source files
0 errors, 0 warnings, 0 infos 
Completed in 48.921sec
Performing static analysis using pyright DONE
Error: Process completed with exit code 1.
```

**Comment by MarcoGorelli** [comment]
> It seems like there is a check that keeps failing `CI > Checks (pull_request) > Typing validation`. I do not understand the error report so I'm not sure what is going wrong. Please see the details of the error report below and advise:
> 
> ```
> Run ci/code_checks.sh typing
> mypy --version
> mypy 0.910
> Performing static analysis using mypy
> pandas/core/frame.py:997: error: Incompatible return value type (got "Optional[str]", expected "str")  [return-value]
> Found 1 error in 1 file (checked 1318 source files)
> Performing static analysis using mypy DONE
> Performing static analysis using pyright
> No configuration file found.
> pyproject.toml file found at /home/runner/work/pandas/pandas.
> Loading pyproject.toml file at /home/runner/work/pandas/pandas/pyproject.toml
> Assuming Python platform Linux
> Searching for source files
> Found 311 source files
> 0 errors, 0 warnings, 0 infos 
> Completed in 48.921sec
> Performing static analysis using pyright DONE
> Error: Process completed with exit code 1.
> ```

It's unrelated, and fixed on master, don't worry about it

**Comment by ruankie** [comment]
> It's unrelated, and fixed on master, don't worry about it

Ok great. Is there anything else I can do regarding this PR before it can be merged?

**Comment by MarcoGorelli** [comment]
Looks good to me, I'll just leave it open a bit in case others have comments

**Comment by ruankie** [comment]
Good idea, thanks @MarcoGorelli

File: pandas/core/generic.py [hunk 1]
```diff
@@ -11221,6 +11221,34 @@ def _doc_params(cls):
...
+...                  ).set_index('person_id')
+>>> df
+           age  height
+person_id
+0           21    1.61
+1           25    1.87
+2           62    1.49
+3           43    2.01
+
+The variance of the columns can be found as follows:
```
Comment (mroeschke) [inline pandas/core/generic.py:11240]: Nit: I think this sentence can be removed (somewhat obvious from the beginning of the doc string)

File: pandas/core/generic.py [hunk 2]
```diff
@@ -11221,6 +11221,34 @@ def _doc_params(cls):
...
+
+>>> df.var()
+age       352.916667
+height      0.056367
+
+Alternatively, `ddof=0` can be set to normalize by N instead of N-1:
```
Comment (mroeschke) [inline pandas/core/generic.py:11246]: Nit: I think double back ticks ``` `` ``` would look better around ddof=0

**Comment by ruankie** [comment]
Thanks, @mroeschke 
Just made the changes you recommended.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @ruankie

**Pull request merged**